### DISCUSSION TITLE
Failing to build custom otelcol

### DISCUSSION THREAD

**Discussion by ljulliar** [open-telemetry/opentelemetry-collector-contrib#17876] [Q&A] [answered]
Hi all,
I'm using the ocb tool to build my custom otelcol. I want a minimal version that collect local host metrics and send them to Cloud Logging on Google Cloud.

My builder.yaml is attached 
[otelcol-builder.yaml.txt](https://github.com/open-telemetry/opentelemetry-collector-contrib/files/10465250/otelcol-builder.yaml.txt)

The builder won't succeed and I got the erro message below. I suspect this is a version related problem. I tried to introduce various entries in the replaces section of the builder file but to no avail. Any suggestion or workaround would be much appreciated.

``` 
Error: failed to update go.mod: exit status 1. Output:
go: finding module for package go.opentelemetry.io/collector/service/servicetest
go: finding module for package go.opentelemetry.io/collector/config
go.opentelemetry.io/collector/cmd/builder imports
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter imports
	go.opentelemetry.io/collector/config: module go.opentelemetry.io/collector@latest found (v0.69.1), but does not contain package go.opentelemetry.io/collector/config
go.opentelemetry.io/collector/cmd/builder imports
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter tested by
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter.test imports
	go.opentelemetry.io/collector/service/servicetest: module go.opentelemetry.io/collector@latest found (v0.69.1), but does not contain package go.opentelemetry.io/collector/service/servicetest
```

**Comment by jpkrohling** [comment]
Change your contrib versions to v0.65.0, and make sure you are using the latest ocb version.

**Reply by ljulliar** [reply]
Thank you for your reply. I'm not sure i understand the change you suggested. The contrib modules I'm using are already declared as v0.65.0 in the builder yaml file. See you must mean something different.

I'm already running the lates ocb version (0.69.1)

**Reply by ljulliar** [reply]
For now I managed to build something with everything v0.57.2. 
v0.64.0 and 0.64.1 fail too.

**Reply by jpkrohling** [reply]
Sorry, I meant v0.69.0. Basically, you should keep the same versions for everything: ocb, core, contrib. This latest release is a bit different, as core is one patch version ahead: 0.69.1, where everything else is at 0.69.0.

**Reply by ljulliar** [reply]
FYI I opened an issue at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17899
and it seems that there are broken depedencies in the newer versions of the builder.

**Comment by codeboten** [comment, marked as answer]
The opened issue has been resolved, I believe this should work as of v0.70.0
